snap-update-tools/get_version_store_snap.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+"""
+This script is ment to query the store to get the latest version published
+of a given snap
+
+Example invocation:
+    $ python3 get_latest_snap_published.py checkbox beta
+    v3.1.2-dev20
+"""
+import sys
+
+from argparse import ArgumentParser
+
+from utils import get_snap_info_from_store
+
+
+def parse_args(argv):
+    parser = ArgumentParser()
+    parser.add_argument("snap_name", help="Name of the snap")
+    parser.add_argument(
+        "channel",
+        choices=["edge", "beta", "stable"],
+        help="Channel to get the snap from",
+    )
+
+    return parser.parse_args(argv)
+
+
+def get_latest_version(snap_info, channel):
+    for entry in snap_info["channel-map"]:
+        if entry["channel"]["name"] == channel:
+            return entry["version"]
+    raise SystemExit("No version found in the specified channel")
+
+
+def get_version_store_snap(snap_name: str, channel: str):
```

Comment:
It takes a while to understand the function name.

The channel just flies through the function, and the caller already has whatever is returned, so I don't see the value in this arg.

Reply:
The arg is there because that is the main functionality function. This is intended so that one is able to use the function without injecting+parsing the args. I have done this in all tools that I have developed for the release process, it is pretty convenient. You are right about the name, I have changed it, now it should flow way better!